ckanext/canada/view.py

```diff
@@ -579,6 +596,79 @@ def ckanadmin_publish_datasets():
     return h.redirect_to('canada.ckanadmin_publish')
 
 
+@canada_views.route('/ckan-admin/job-queue', methods=['GET'])
+def ckanadmin_job_queue():
+    """
+    Lists all of the queued and running jobs.
+    """
+    try:
+        jobs = get_action('job_list')({}, {})
+    except NotAuthorized:
+        abort(403, _('Not authorized to see this page'))
+
+    warning = False
+    if jobs and datetime.strptime(jobs[0]['created'], '%Y-%m-%dT%H:%M:%S') < (datetime.now() - timedelta(minutes=18)):
+        warning = True
+
+    job_list = []
+    mystery_job_count = 0
+    for job in jobs[:25]:
+        try:
+            job_obj = Job.fetch(job.get('id'))
+            job_args = job_obj.args[0]
+        except (NoSuchJobError, KeyError):
+            mystery_job_count += 1
+            continue
+
+        if '.run_validation_job' in job_obj.func_name:
```

Comment:
this is kind of unpleasant. maybe if it's a core feature we could add labels to the jobs and use a helper to provide human-readable descriptions so plugins can add their own human-readable versions

Reply:
Redis job queue does support metadata titles, either CKAN does not use them, or Xloader and Validation is not using them. So we can just get the job object from `rq` and print out the title.

And then I guess we can just add those translations into our `strings.py`

Reply:
So yeah, you can just pass a `title` attribute to the enqueue toolkit method and it will add the meta title.

Hmmm right. Well we will still need to do some nasty conditions because the job data is not the same for everything, and we want the Resource information.

@wardi let me know if you have any ideas on how to make this not ugly and a bit better

But the new code here would required these two PRs:
- https://github.com/open-data/ckanext-xloader/pull/17
- https://github.com/open-data/ckanext-validation/pull/10

Reply:
@wardi semantics are in, but as we still need to have the if/else statements to be able to get the Resource info, it might just be better to condition on the callback method?